onnxruntime/core/platform/windows/env.cc

```diff
@@ -76,9 +77,26 @@ class WindowsThread : public EnvThread {
   typedef HRESULT(WINAPI* SetThreadDescriptionFunc)(HANDLE hThread, PCWSTR lpThreadDescription);
   static unsigned __stdcall ThreadMain(void* param) {
     std::unique_ptr<Param> p((Param*)param);
-    // TODO: should I try to use SetThreadSelectedCpuSets?
-    if (!p->thread_options.affinity.empty())
-      SetThreadAffinityMask(GetCurrentThread(), p->thread_options.affinity[p->index]);
+    const auto& affinities = p->thread_options.affinity;
+    size_t num_affinity = affinities.size();
+    size_t offset = static_cast<size_t>(p->index) * 2;
```

Comment:
>static_cast<size_t> [](http://example.com/codeflow?start=20&length=19)

What prevents us using the type that is p-->index?

Reply:
index is an int, we will get a warning without that static-cast